optimade/server/entry_collections/mongo.py

```diff
@@ -113,18 +113,24 @@ def _run_db_query(
 
         """
 
-        results = []
-        for doc in self.collection.find(**criteria):
-            if criteria.get("projection", {}).get("_id"):
-                doc["_id"] = str(doc["_id"])
-            results.append(doc)
+        # Handle MongoDB ObjectIDs:
+        # - If they were not requested, then explicitly remove them
+        # - If they were requested, then cast them to strings in the response
+        if "_id" not in criteria.get("projection", {}):
+            criteria["projection"]["_id"] = False
+
+        if criteria.get("projection", {}).get("_id"):
+            criteria["projection"]["_id"] = {"$toString": "$_id"}
```

Comment:
Did you test this line ? 
according to Codecov it is not visited in any test.

Reply:
Good point, I tested it locally with a real MongoDB but it looks like this projection isn't implemented in mongomock, will have to add a conditional.

We can also update our test config to use the MongoDB `_id` field as the OPTIMADE `immutable_id` which will mean this gets hit in the tests.

Reply:
(moved this to another PR, see #1142)